CommonConcepts/Plugins/Rhetos.Dsl.DefaultConcepts/DataStructure/MethodConceptInfo.cs

```diff
@@ -0,0 +1,33 @@
+﻿/*
+    Copyright (C) 2014 Omega software d.o.o.
+
+    This file is part of Rhetos.
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+using System.ComponentModel.Composition;
+
+namespace Rhetos.Dsl.DefaultConcepts
+{
+    [Export(typeof(IConceptInfo))]
+    [ConceptKeyword("Method")]
+    public class MethodConceptInfo : IConceptInfo
+    {
+        [ConceptKey]
+        public DataStructureInfo DataStructure { get; set; }
+        [ConceptKey]

```

Comment:
We should avoid having code snippets (MethodImplementation) in ConceptKey, because it makes referencing this concept very difficult, and any change in implementation will change the feature's identification.

This kind of concept should have a string property `Name` (to serve as a business rule name), as a part of the ConceptKey. This will allow other concepts in future to reference this feature in DSL scripts if needed. Add a validation that Name should be an identifier (DslUtility.ValidateIdentifier), as a standard for business rule names.